src/enzyme_ad/jax/Passes/EnzymeHLOOpt.cpp

```diff
@@ -7792,6 +7792,231 @@ struct TransposeReduceSimplify : public OpRewritePattern<stablehlo::ReduceOp> {
   }
 };
 
+// (select (x > 0) z (neg z)) -> (mul (sign x) z)
+// (select (x >= 0) z (neg z)) -> (mul (sign x) z)
+// (select (x > 0) (neg z) z) -> (mul (sign x) (neg z))
+// (select (x >= 0) (neg z) z) -> (mul (sign x) (neg z))
```

Comment:
is this actually simpler/faster?

Reply:
This one mostly enables the other optimizations.